### ISSUE TITLE
format of zotero:// links

### ISSUE THREAD

**Issue by damiencourousse** [egh/zotxt-emacs#53]
Reading from the source code, I understand that this extension uses the following format for zotero:// links:
- with zotero cite keys: `zotero://select/items/` + `key`

However, exports from better bibtex can generate links with two different formats, as follows (see [here](https://github.com/retorquere/zotero-better-bibtex/blob/275d09b059617fa35febc06289a44c2b388c9dd8/translators/Better%20BibTeX%20Citation%20Key%20Quick%20Copy.ts#L10))
- with the citation key: `zotero://select/items/` + `citekey`
- with the zotero reference key: `zotero://select/library/items` + `key`

However, I could not find a specification of the different link formats, so I don't know how to handle this inconsistency.

Currently, if i drag'n drop a reference from Zotero into an org document, with better bibtex, to generate a zotero:// link, I cannot open those links with zotxt-emacs.  I use zotxt-emacs to open zotero:// links, and I would like to have the possibility to use both better bibtex and zotxt-emacs to insert links to references.

**Comment by egh** [comment]
Hi @damiencourousse ! Sorry for the late response. I think this is mistake in the BBT export. It should look like `zotero://select/items/[LIBRARY]_[KEY]`. This was an internal format used by Zotero itself - it may still be. I don't use the translators - I insert links directly from emacs - so I hadn't seen this issue.

**Comment by damiencourousse** [comment]
Thanks for your feedback.  I have opened an issue in BBT.  Feel free to comment!

**Comment by damiencourousse** [comment]
You are right.  The current format is `zotero://select/items/[LIBRARY]_[KEY]`.  It was a misunderstanding from me, since on my setting, `[LIBRARY]` is `1`.

As reported in the BBT issue, this format is still supported but deprecated.  Do you plan updating the format of zotero links in zotxt-emacs?

**Comment by egh** [comment]
> As reported in the BBT issue, this format is still supported but deprecated. Do you plan updating the format of zotero links in zotxt-emacs?

Ah, I didn't know that! I will fix this. Thank you.

**Comment by brunoperdigao** [comment]
Hi, I hope it is related to this topic, but I am wondering if zotxt can handle zotero links created by zotfile. Their format is `zotero://open-pdf/library/items/`. These links open the pdf in a specific page, related to the annotations.
I tried to open with the `org-zotxt-open-attachment` command but It didn't work.

**Comment by egh** [comment]
@brunoperdigao Good question! Right now it can't, but it would be very easy to implement. I assume the item key follows after `/items/`?

**Comment by brunoperdigao** [comment]
@egh Yes, the item key and the page number, which directs to the specific page, for example `zotero://open-pdf/library/items/XXXXXXXX?page=3`
I tried to find more information on their documentation, but they don't go into much detail about it: http://zotfile.com/#extract-pdf-annotations
It would be great to have this implemented! Thanks

**Comment by zzeitt** [comment]
I had the similar issue on the iOS platform. 
It seems like the link generated by zotxt cannot be recognized by Zotero iOS. 
However, links formatted like : `zotero://select/library/items/XXXXXXXX` is valid (you can jump to the expected item after click the link).
> Infomation about the Zotero iOS link/url processing could be found here: https://github.com/zotero/zotero-ios/blob/5d0816061a5ed9849c82ac8ca087c4dc2ee489d2/Zotero/Controllers/CustomURLController.swift#L37

So I wonder what is the status of the link format upgrading plan?
